rts/lra/test/basic/src/test/java/io/narayana/lra/arquillian/api/CoordinatorApi_1_0_IT.java

```diff
@@ -0,0 +1,821 @@
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+import org.junit.runner.RunWith;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.GenericType;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Link;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URLDecoder;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.emptyCollectionOf;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.lessThan;
+import static org.hamcrest.core.IsCollectionContaining.hasItem;
+import static org.hamcrest.core.IsCollectionContaining.hasItems;
+import static org.hamcrest.core.IsNot.not;
+
+/**
+ * REST API tests against the LRA coordinator in version for LRA 1.0.
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class CoordinatorApi_1_0_IT {
+    static final String API_VERSION_1_0 = "1.0";
+
+    private Client client;
+    private NarayanaLRAClient lraClient;
+    private String coordinatorUrl;
+    private List<URI> lrasToAfterFinish;
+
+    static final String NOT_SUPPORTED_FUTURE_LRA_VERSION = "42.1";
+
+    static final String LRA_API_VERSION_HEADER_NAME_V1_0 = "Narayana-LRA-API-version";
+    static final String RECOVERY_HEADER_NAME_V1_0 = "Long-Running-Action-Recovery";
+    static final String STATUS_PARAM_NAME_V1_0 = "Status";
+    static final String CLIENT_ID_PARAM_NAME_V1_0 = "ClientID";
+    static final String TIME_LIMIT_PARAM_NAME_V1_0 = "TimeLimit";
+    static final String PARENT_LRA_PARAM_NAME_V1_0 = "ParentLRA";
+
+    @Rule
+    public TestName testName = new TestName();
+
+    @Deployment
+    public static WebArchive deploy() {
+        return Deployer.deploy(CoordinatorApi_1_0_IT.class.getSimpleName());
+    }
+
+    @Before
+    public void before() {
+        client = ClientBuilder.newClient();
+        lraClient = new NarayanaLRAClient();
+        coordinatorUrl = lraClient.getCoordinatorUrl();
+        lrasToAfterFinish = new ArrayList<>();
+    }
+
+    @After
+    public void after() {
+        for (URI lraToFinish: lrasToAfterFinish) {
+            lraClient.cancelLRA(lraToFinish);
+        }
+        if (client != null) {
+            client.close();
+        }
+    }
+
+    @Test
+    public void getAllLRAs() {
```

Comment:
I cannot decipher what this test is doing so I didn't review it (same for the remaining tests in this class). I assume it is something to do with API conformance (although we have loads of tests already for spec compliance so it's not immediately obvious what the extra tests are verifying, is there any duplication going on?). You often ask me for better test documentation so it would be courteous for you to follow your own advice. Minimally you could add a comment above the class giving a general overview of the goal and what a typical test in the suite is doing.

Reply:
It's quite dense code and not easy to understand what it is testing but okay.

Reply:
I think REST Assured could help to make it clearer (but unfortunately I started to think about it late during the development). What do you think, would that help?
I may create an issue on me to rewrite the tests with rest-assured to be more readable.

Reply:
I'm not sure rewriting the tests using a different API will help make the code more readable. But leave it as it is and whoever has to write the next set of tests for a new version will just have to figure out what it's doing.

Reply:
Ok. If you have some proposal how to make it more readable it would be nice.

These tests needs to check multiple things - returned status, headers, content, and data. There are many asserts for it. I was thinking if the API designed for such type of things could help. Not sure though.
I'm resolving this for now. But please reopen if I did not address some point.